### PR TITLE
fixes #171 - Correcting documentation regarding requirements

### LINKED ISSUE #171: Solar fails after initial vagrant up
zen@nandrolone  ~/projects/Mirantis/solar   master  git status
On branch master
Your branch is up-to-date with 'origin/master'.
nothing to commit, working directory clean
 zen@nandrolone  ~/projects/Mirantis/solar   master  vagrant up
Bringing machine 'solar-dev' up with 'virtualbox' provider...
Bringing machine 'solar-dev1' up with 'virtualbox' provider...
Bringing machine 'solar-dev2' up with 'virtualbox' provider...
==> solar-dev: Box 'cgenie/solar-master' could not be found. Attempting to find and install...
    solar-dev: Box Provider: virtualbox
    solar-dev: Box Version: >= 0
==> solar-dev: Loading metadata for box 'cgenie/solar-master'
    solar-dev: URL: https://atlas.hashicorp.com/cgenie/solar-master
==> solar-dev: Adding box 'cgenie/solar-master' (v1.1.3) for provider: virtualbox
    solar-dev: Downloading: https://atlas.hashicorp.com/cgenie/boxes/solar-master/versions/1.1.3/providers/virtualbox.box
==> solar-dev: Successfully added box 'cgenie/solar-master' (v1.1.3) for 'virtualbox'!
==> solar-dev: Importing base box 'cgenie/solar-master'...
==> solar-dev: Matching MAC address for NAT networking...
==> solar-dev: Checking if box 'cgenie/solar-master' is up to date...
==> solar-dev: Setting the name of the VM: solar-dev
==> solar-dev: Clearing any previously set forwarded ports...
==> solar-dev: Clearing any previously set network interfaces...
==> solar-dev: Preparing network interfaces based on configuration...
    solar-dev: Adapter 1: nat
    solar-dev: Adapter 2: hostonly
==> solar-dev: Forwarding ports...
    solar-dev: 22 => 2222 (adapter 1)
==> solar-dev: Running 'pre-boot' VM customizations...
A customization command failed:

["modifyvm", :id, "--memory", 1024, "--cpus", 1, "--paravirtprovider", "kvm", "--ioapic", "on"]

The following error was experienced:

#<Vagrant::Errors::VBoxManageError: There was an error while executing `VBoxManage`, a CLI used by Vagrant
for controlling VirtualBox. The command and stderr is shown below.

Command: ["modifyvm", "eafc76f9-4520-4eff-9018-ec6a478debbe", "--memory", "1024", "--cpus", "1", "--paravirtprovider", "kvm", "--ioapic", "on"]

Stderr: Oracle VM VirtualBox Command Line Management Interface Version 4.3.30
(C) 2005-2015 Oracle Corporation
All rights reserved.

Usage:

VBoxManage modifyvm <uuid|vmname>
                            [--name <name>]
                            [--groups <group>, ...]
                            [--description <desc>]
                            [--ostype <ostype>]
                            [--iconfile <filename>]
                            [--memory <memorysize in MB>]
                            [--pagefusion on|off]
                            [--vram <vramsize in MB>]
                            [--acpi on|off]
                            [--pciattach 03:04.0]
                            [--pciattach 03:04.0@02:01.0]
                            [--pcidetach 03:04.0]
                            [--ioapic on|off]
                            [--hpet on|off]
                            [--triplefaultreset on|off]
                            [--hwvirtex on|off]
                            [--nestedpaging on|off]
                            [--largepages on|off]
                            [--vtxvpid on|off]
                            [--vtxux on|off]
                            [--pae on|off]
                            [--longmode on|off]
                            [--synthcpu on|off]
                            [--cpuidset <leaf> <eax> <ebx> <ecx> <edx>]
                            [--cpuidremove <leaf>]
                            [--cpuidremoveall]
                            [--hardwareuuid <uuid>]
                            [--cpus <number>]
                            [--cpuhotplug on|off]
                            [--plugcpu <id>]
                            [--unplugcpu <id>]
                            [--cpuexecutioncap <1-100>]
                            [--rtcuseutc on|off]
                            [--graphicscontroller none|vboxvga|vmsvga]
                            [--monitorcount <number>]
                            [--accelerate3d on|off]
                            [--accelerate2dvideo on|off]
                            [--firmware bios|efi|efi32|efi64]
                            [--chipset ich9|piix3]
                            [--bioslogofadein on|off]
                            [--bioslogofadeout on|off]
                            [--bioslogodisplaytime <msec>]
                            [--bioslogoimagepath <imagepath>]
                            [--biosbootmenu disabled|menuonly|messageandmenu]
                            [--biossystemtimeoffset <msec>]
                            [--biospxedebug on|off]
                            [--boot<1-4> none|floppy|dvd|disk|net>]
                            [--nic<1-N> none|null|nat|bridged|intnet|hostonly|
                                        generic|natnetwork]
                            [--nictype<1-N> Am79C970A|Am79C973|
                                            82540EM|82543GC|82545EM|
                                            virtio]
                            [--cableconnected<1-N> on|off]
                            [--nictrace<1-N> on|off]
                            [--nictracefile<1-N> <filename>]
                            [--nicproperty<1-N> name=[value]]
                            [--nicspeed<1-N> <kbps>]
                            [--nicbootprio<1-N> <priority>]
                            [--nicpromisc<1-N> deny|allow-vms|allow-all]
                            [--nicbandwidthgroup<1-N> none|<name>]
                            [--bridgeadapter<1-N> none|<devicename>]
                            [--hostonlyadapter<1-N> none|<devicename>]
                            [--intnet<1-N> <network name>]
                            [--nat-network<1-N> <network name>]
                            [--nicgenericdrv<1-N> <driver>
                            [--natnet<1-N> <network>|default]
                            [--natsettings<1-N> [<mtu>],[<socksnd>],
                                                [<sockrcv>],[<tcpsnd>],
                                                [<tcprcv>]]
                            [--natpf<1-N> [<rulename>],tcp|udp,[<hostip>],
                                          <hostport>,[<guestip>],<guestport>]
                            [--natpf<1-N> delete <rulename>]
                            [--nattftpprefix<1-N> <prefix>]
                            [--nattftpfile<1-N> <file>]
                            [--nattftpserver<1-N> <ip>]
                            [--natbindip<1-N> <ip>
                            [--natdnspassdomain<1-N> on|off]
                            [--natdnsproxy<1-N> on|off]
                            [--natdnshostresolver<1-N> on|off]
                            [--nataliasmode<1-N> default|[log],[proxyonly],
                                                         [sameports]]
                            [--macaddress<1-N> auto|<mac>]
                            [--mouse ps2|usb|usbtablet|usbmultitouch]
                            [--keyboard ps2|usb
                            [--uart<1-N> off|<I/O base> <IRQ>]
                            [--uartmode<1-N> disconnected|
                                             server <pipe>|
                                             client <pipe>|
                                             file <file>|
                                             <devicename>]
                            [--guestmemoryballoon <balloonsize in MB>]
                            [--audio none|null|coreaudio]
                            [--audiocontroller ac97|hda|sb16]
                            [--clipboard disabled|hosttoguest|guesttohost|
                                         bidirectional]
                            [--draganddrop disabled|hosttoguest
                            [--vrde on|off]
                            [--vrdeextpack default|<name>
                            [--vrdeproperty <name=[value]>]
                            [--vrdeport <hostport>]
                            [--vrdeaddress <hostip>]
                            [--vrdeauthtype null|external|guest]
                            [--vrdeauthlibrary default|<name>
                            [--vrdemulticon on|off]
                            [--vrdereusecon on|off]
                            [--vrdevideochannel on|off]
                            [--vrdevideochannelquality <percent>]
                            [--usb on|off]
                            [--usbehci on|off]
                            [--snapshotfolder default|<path>]
                            [--teleporter on|off]
                            [--teleporterport <port>]
                            [--teleporteraddress <address|empty>
                            [--teleporterpassword <password>]
                            [--teleporterpasswordfile <file>|stdin]
                            [--tracing-enabled on|off]
                            [--tracing-config <config-string>]
                            [--tracing-allow-vm-access on|off]
                            [--usbcardreader on|off]
                            [--autostart-enabled on|off]
                            [--autostart-delay <seconds>]
                            [--vcpenabled on|off]
                            [--vcpscreens [<display>],...
                            [--vcpfile <filename>]
                            [--vcpwidth <width>]
                            [--vcpheight <height>]
                            [--vcprate <rate>]
                            [--vcpfps <fps>]
                            [--defaultfrontend default|<name>]

VBoxManage: error: Unknown option: --paravirtprovider
>

Please fix this customization and try again.

### PR DESCRIPTION
**Pull request by zen**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,27 @@
...
+
+Linux or MacOS
+
+## Additional software
+
+[VirtualBox](https://www.virtualbox.org/wiki/Downloads): 5.x
+
+[Vagrant](http://www.vagrantup.com/downloads.html): 1.7.x
+
+Note: You can also use Vagrant 4.x but you will need to uncomment paravirtprovider setting in vagrant-settings.yaml:
```
Comment (loles) [inline README.md]: You should uncomment it for 5.0. In 4.x it will not work

File: README.md [hunk 2]
```diff
@@ -1,8 +1,27 @@
...
+```bash
```
Comment (CGenie) [inline README.md]: This is `yaml` not `bash`.

File: README.md (hunk 1 above)
Comment (CGenie) [inline README.md]: You should quote file names: `vagrant-settings.yaml` with backtick.

File: README.md [hunk 3]
```diff
@@ -1,8 +1,27 @@
...
+paravirtprovider: kvm
+```
+
+For details see Customizing vagrant-settings.yaml section.
```
Comment (CGenie) [inline README.md]: Filename quoting.

File: README.md [hunk 4]
```diff
@@ -267,3 +286,8 @@ riak_master_service.connect_list(
 For full Riak example, please look at `examples/riak/riaks-template.py`.
 
 Full documentation of individual functions is found in the `solar/template.py` file.
+
+
+# Customizing vagrant-settings.yaml
+
+Solar is shipped with sane defaults in vagrant-setting.yaml_defaults. If you need to adjust them for your needs, e.g. changing resource allocation for VirtualBox machines, you should just compy the file to vagrant-setting.yaml and make your modifications.
```
Comment (CGenie) [inline README.md]: ditto

**Pull request merged**